ci/test-appimagetool.sh

```diff
@@ -117,3 +117,21 @@ if [ "$hash1" != "$hash2" ]; then
     echo "Hash $hash1 doesn't match hash $hash2!"
     exit 1
 fi
+
+log "check --mksquashfs-opt forwarding"
+out=$("$appimagetool" appimagetool.AppDir appimagetool.AppImage --mksquashfs-opt "-misspelt-option" 2>&1)
+echo "${out}" | grep -q "invalid option"
+"$appimagetool" appimagetool.AppDir appimagetool.AppImage.1
+"$appimagetool" appimagetool.AppDir appimagetool.AppImage.2 --mksquashfs-opt "-mem" --mksquashfs-opt "100M"
+"$appimagetool" appimagetool.AppDir appimagetool.AppImage.3 --mksquashfs-opt "-all-time" --mksquashfs-opt "12345"
+hash1=$(sha256sum appimagetool.AppImage.1 | awk '{print $1}')
+hash2=$(sha256sum appimagetool.AppImage.2 | awk '{print $1}')
+hash3=$(sha256sum appimagetool.AppImage.3 | awk '{print $1}')
+if [ "$hash1" != "$hash2" ]; then
+    echo "Hashes of regular and mem-restricted AppImages differ"
```

Comment:
I don't quite like the method used in this check, but I guess you can't check `-mem` otherwise, so this is okay. Should catch most future issues.

It's good you call those checks, not tests.